packages/picasso.js/src/core/component/component-factory.js

```diff
@@ -489,6 +491,7 @@ function componentFactory(definition, context = {}) {
     });
 
     if (currentNodes && settings.animations && settings.animations.enabled) {
+      settings.animations.compensateForLayoutChanges(currentNodes, instanceContext.rect.computed, preComputedRect);
```

Comment:
First of all - don't we need to check if the `compensateForLayoutChanges` function exist before we call it here?

Then, to make it a little easier to work with, we rename the input parameters and send them in as an object, something like this:
```suggestion
      settings.animations.compensateForLayoutChanges({currentNodes, currentRect: instanceContext.rect.computed, newRect: preComputedRect});
```

Reply:
good points!